build/deepcopy.bzl

```diff
@@ -0,0 +1,52 @@
+_cmd = """
+  # location of prebuilt deepcopy generator
+  dcg=$$PWD/'$(location //vendor/k8s.io/code-generator/cmd/deepcopy-gen)'
+  # original pwd
+  export O=$$PWD
+  # gopath/goroot for genrule
+  export GOPATH=$$PWD/.go
+  export GOROOT=/usr/lib/google-golang
+
+  # symlink in source into new gopath
+  mkdir -p $$GOPATH/src/k8s.io
+  ln -snf $$PWD $$GOPATH/src/k8s.io/kubernetes
+  # symlink in all the staging dirs
+  for i in $$(ls staging/src/k8s.io); do
+    ln -snf $$PWD/staging/src/k8s.io/$$i $$GOPATH/src/k8s.io/$$i
```

Comment:
or do you mean that all of the staging dirs need to be installed in `$GOPATH/src/k8s.io/`, rather than in vendor?

Reply:
Yes, although TBH I'm not really sure if this is necessary.

Between our staging pretend repo, symlinks galore and golang's vendoring logic I have no idea what it is supposed to look like. Solution: create symlinks until it works and distill later 😈